### ISSUE TITLE
mergify/bors-like merging process

### ISSUE THREAD

**Issue by Zimmi48** [coq/bot#18]
This would ensure better reliability because CI would run on commits before they become the `HEAD` of `master` or a stable branch.

This would however require first a (planned) change to the way overlays are handled (cf. coq/coq#6724).

There is also a question on how to allow the `pkg:nix` job to upload its result to Cachix (the `CACHIX_SIGNING_KEY` secret variable is only available on protected branches, but it could also be passed to a triggered pipeline).

More generally, do we want the staging branch to be protected (in which case there will be a number of staging branches) or is it OK to force-push to staging? Do we need the staging branch to appear on the GitHub repository or can it be only pushed to GitLab? If we could gather all of Coq CI on GitLab (we are not so far), then there would be no point in pushing to the GitHub repository (there is also the option of triggering external CI from GitLab CI).

**Comment by Zimmi48** [comment]
A good first step would be to make the merging process always go through @coqbot.

**Comment by Zimmi48** [comment]
A good first step even before this one is to allow any code owner to merge through @coqbot.

**Comment by ejgallego** [comment]
Sounds good! Tho we will have to become way more stricter in issues such as coq/coq#8533

**Comment by Zimmi48** [comment]
Yep, indeed. To go the full way to bors-like process, getting a very reliable CI is a prerequisite.
